frontend/src/AppHelpers.res

```diff
@@ -15,3 +15,51 @@ let jobUrl = (~lines=?, jobId) => {
 }
 
 let defaultBenchmarkName = "default"
+
+// Get sorted (by runAt) metadata information for all commits
+let sortedMetadata = dataByTestName => {
+  dataByTestName
+  ->Belt.Map.String.valuesToArray
+  ->Belt.Array.map(((_, dataByMetricName)) => dataByMetricName->Belt.Map.String.valuesToArray)
+  ->Belt.Array.concatMany
+  ->Belt.Array.map(((_, md)) => md)
+  ->Belt.Array.concatMany
+  ->Belt.Array.reduce(Belt.Map.String.empty, (acc, md) => {
+    Belt.Map.String.set(acc, md["commit"], md)
+  })
+  ->Belt.Map.String.valuesToArray
+  ->Belt.SortArray.stableSortBy((a, b) =>
+    compare(Js.Date.getTime(a["runAt"]), Js.Date.getTime(b["runAt"]))
+  )
+}
+
+// Fill in NaN values and add missing commit metadata
+let addMissingCommits = ((metricTimeseries, metricMetadata) as data, commitMetadata) => {
+  let n = Belt.Array.length(commitMetadata)
+  switch Belt.Array.length(metricTimeseries) < n {
+  | false => data
+  | true => {
+      let commits = commitMetadata->Belt.Array.map(m => m["commit"])
+      let filledTimeseries = Belt.Array.make(n, [Js.Float._NaN, Js.Float._NaN, Js.Float._NaN])
+      ignore(
+        metricMetadata->Belt.Array.mapWithIndex((idx, m) => {
+          let idx_ = commits->Belt.Array.getIndexBy(c => c == m["commit"])
+          switch idx_ {
+          | Some(i) => Belt.Array.set(filledTimeseries, i, Belt.Array.getExn(metricTimeseries, idx))
+          | _ => false
+          }
```

Comment:
`filledTimeseries` could be defined as the result of the `mapWithIndex`, to skip the `make/set/ignore` :)

Reply:
Yes, this function is adding a `NaN` value for the commits which are missing for a particular metric. So, `commitMetadata` which has the metadata for all commits, and `metricMetadata` which has metadata for only the commits for that particular metric are going to have different lengths.

Reply:
Yep! Pushed a commit to make this code more understandable. Thanks! :)